src/main/java/core/basesyntax/data/DataConverterImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.data;
+
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.List;
+import java.util.Objects;
+
+public class DataConverterImpl implements DataConverter {
+    private Formater formater;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(s -> {
+                    if (formater == null) {
+                        formater = new FormaterImpl(s);
+                        return null;
+                    }
+                    return parseFromString(s);
+                })
+                .filter(Objects::nonNull)
```

Comment:
Using `Objects::nonNull` to filter out null values suggests that nulls are expected from 'parseFromString', which should not be the case. Ensure 'parseFromString' returns a valid `FruitTransaction` or throws an exception if it cannot parse the input.